vcf-dumper/vcf-dumper-ws/src/main/java/uk/ac/ebi/eva/vcfdump/server/HtsgetVcfController.java

```diff
@@ -0,0 +1,235 @@
+            @RequestParam(name = "end", required = false) Integer end,
+            @RequestParam(name = "fields", required = false) List<String> fields,
+            @RequestParam(name = "tags", required = false, defaultValue = "") String tags,
+            @RequestParam(name = "notags", required = false, defaultValue = "") String notags,
+            HttpServletRequest request,
+            HttpServletResponse response)
+            throws IllegalAccessException, IllegalOpenCGACredentialsException, InstantiationException, IOException,
+            StorageManagerException, URISyntaxException, ClassNotFoundException {
+
+        if (!VCF.equals(format)) {
+            return ResponseEntity.status(HttpStatus.BAD_REQUEST)
+                    .body(new HtsGetError("UnsupportedFormat", "Specified format is not supported by this server"));
+        }
+
+        String dbName = "eva_" + species;
+        MultivaluedMap<String, String> queryParameters = new MultivaluedHashMap<>();
+        queryParameters.put(VariantDBAdaptor.REGION, Collections
+                .singletonList(referenceName));
+
+        int blockSize = Integer.parseInt(evaProperties.getProperty("eva.htsget.blocksize"));
+
+        VariantExporterController controller = new VariantExporterController(dbName, Arrays.asList(id.split(",")),
+                                                                             evaProperties,
+                                                                             queryParameters, blockSize);
+        if (!controller.validateSpecies()) {
+            return ResponseEntity.status(HttpStatus.BAD_REQUEST)
+                    .body(new HtsGetError("InvalidInput", "The requested species is not available"));
+        }
+        if (!controller.validateStudies()) {
+            return ResponseEntity.status(HttpStatus.BAD_REQUEST)
+                    .body(new HtsGetError("InvalidInput", "The requested study(ies) is not available"));
+        }
+
+        if (start == null) {
+            start = controller.getCoordinateOfFirstVariant(referenceName);
+        }
+        if (end == null) {
+            end = controller.getCoordinateOfLastVariant(referenceName);
+        }
+
+        if (end <= start) {
+            return ResponseEntity.status(HttpStatus.BAD_REQUEST)
+                    .body(new HtsGetError("InvalidRange", "The requested range cannot be satisfied"));
+        }
+
+        List<Region> regionList = controller.divideChromosomeInChunks(referenceName, start, end);
+
+        HtsGetResponse htsGetResponse = buildHtsGetResponse(id, referenceName, species, request, regionList);
+        return ResponseEntity.status(HttpStatus.OK).body(htsGetResponse);
+
+    }
+
+    private HtsGetResponse buildHtsGetResponse(String id, String referenceName, String species,
+                                               HttpServletRequest request, List<Region> regionList) {
+        HtsGetResponse htsGetResponse = new HtsGetResponse();
```

Comment:
If a response is only valid when the format and the URLs are defined, it is better to have a constructor that assigns those values, instead of doing it via setters. In fact, those setters can be deleted if they are not used after the object has been constructed.